packages/types/src/types.ts

```diff
@@ -154,6 +154,11 @@ export interface SignatureOptions {
 
 export type ArgsDef = Record<string, Constructor>;
 
+// A type alias for [Type1, Type2] & Codec, representing a tuple (Type1, Type2)
+// FIXME Implement this generic <Sub> on Tuple.ts itself.
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
+export type ITuple<Sub extends Codec[]> = Sub & Codec
```

Comment:
Introduced this to replace `[Type1, Type2] & Codec`, since I cannot `interface extends ([Type1, Type2] & Codec)`

But as written, should be on Tuple itself ideally. Not straightforward though, as `Tuple extends Array<T>`